doc/examples/affine_registration_3d.py

```diff
@@ -0,0 +1,236 @@
+regtools.overlay_slices(static, warped, None, 1, "Static", "Warped", "warped_trans_1.png")
+regtools.overlay_slices(static, warped, None, 2, "Static", "Warped", "warped_trans_2.png")
+
+"""
+.. figure:: warped_trans_0.png
+   :align: center
+.. figure:: warped_trans_1.png
+   :align: center
+.. figure:: warped_trans_2.png
+   :align: center
+
+   **Registration result by just translating the moving image, using Mutual Information**.
+"""
+
+"""
+Now lets refine with a rigid transform (this may even modify our previously found
+optimal translation)
+"""
+
+transform = regtransforms[('RIGID', 3)]
+params0 = None
+starting_affine = trans
+rigid = affreg.optimize(static, moving, transform, params0,
+                        static_grid2space, moving_grid2space,
+                        starting_affine=starting_affine)
+
+"""
+This produces a slight rotation, and the images are now better aligned
+"""
+
+warped = transform_image(static, static_grid2space, moving, moving_grid2space, rigid)
+regtools.overlay_slices(static, warped, None, 0, "Static", "Warped", "warped_rigid_0.png")
+regtools.overlay_slices(static, warped, None, 1, "Static", "Warped", "warped_rigid_1.png")
+regtools.overlay_slices(static, warped, None, 2, "Static", "Warped", "warped_rigid_2.png")
+
+"""
+.. figure:: warped_rigid_0.png
+   :align: center
+.. figure:: warped_rigid_1.png
+   :align: center
+.. figure:: warped_rigid_2.png
+   :align: center
+
+   **Registration result with a rigid transform, using Mutual Information**.
+"""
+
+"""
+Finally, lets refine with a full affine transform (translation, rotation, scale and
+shear), it is safer to fit more degrees of freedom now, since we must be very close
+to the optimal transform
+"""
+
+transform = regtransforms[('AFFINE', 3)]
+params0 = None
+starting_affine = rigid
+affine = affreg.optimize(static, moving, transform, params0,
+                         static_grid2space, moving_grid2space,
+                         starting_affine=starting_affine)
+
+"""
+This results in a slight shear and scale
+"""
+
+warped = transform_image(static, static_grid2space, moving, moving_grid2space, affine)
+regtools.overlay_slices(static, warped, None, 0, "Static", "Warped", "warped_affine_0.png")
+regtools.overlay_slices(static, warped, None, 1, "Static", "Warped", "warped_affine_1.png")
+regtools.overlay_slices(static, warped, None, 2, "Static", "Warped", "warped_affine_2.png")
+
+"""
+.. figure:: warped_affine_0.png
+   :align: center
+.. figure:: warped_affine_1.png
+   :align: center
+.. figure:: warped_affine_2.png
+   :align: center
+
+   **Registration result with an affine transform, using Mutual Information**.
+
+.. [Mattes03] Mattes, D., Haynor, D. R., Vesselle, H., Lewellen, T. K., & Eubank, W. (2003). PET-CT image registration in the chest using free-form deformations. IEEE Transactions on Medical Imaging, 22(1), 120-8.
+.. [Avants11] Avants, B. B., Tustison, N., & Song, G. (2011). Advanced Normalization Tools ( ANTS ), 1-35.
+
```

Comment:
What you don't show in this tutorial is how to compose two of these.
Like start with a rigid and then do an affine. That would be interesting for many people.